### PR TITLE
Decomp gameflow

### PR DESCRIPTION
**Pull request by rr-**
Decompilation only + a small refactor in gun routines.

Test cases - everything related to the general gameflow:

- Pictures
- FMVs
- Cutscenes
    - Offshore Rig starting animation
- Loading and saving
- Exiting to title
- Dying
- Game start/stop/loop
- Inventory operations:
    - Removing Lara's guns
    - Secret rewards
- Bartoli's Hideout sunset
- Acid pools ("deadly water")
- Floater's ground
- Chicken in Ice Palace
- Level end statistics
- Game end statistics

\+ Lara's guns, especially whether shooting with the armed skidoo works as expected.

Test build: https://home.wind.garden/serve/pub/TR2X-0.2-308-g010c694-Windows.zip

### CODE REVIEW

**Comment by rr-** [comment]
Fixed #158 and #159 in https://github.com/LostArtefacts/TR2X/compare/010c6949845ac1db2f0ad805c156640e838a1f41..34dc3b1807bfd929ac08a11feb1da1a0f5476c41

New test build: https://home.wind.garden/serve/pub/TR2X-0.2-309-g34dc3b1-Windows.zip

**Comment by aredfan** [comment]
@rr- Could you please explain what Floater's ground is?

**Comment by lahm86** [comment]
> @rr- Could you please explain what Floater's ground is?

When Lara falls off into the abyss, she should never reach the floor i.e. there should be no "crunch" on death and it should just go straight to the passport.

**Comment by lahm86** [comment]
Just to note, these gameflow commands are unused AFAIK.
- `Deadly water` is only set in the Ice Palace definition in the script, but it works anywhere there is a lava FD entry in a water room (e.g. Diving Area acid). My guess is they planned it for the script, but then changed the FD handling, as recently done in TR1X.
- `Kill to complete` is set in Home Sweet Home, but removing the GF command has no effect - the counter is triggered by the presence of object ID 262 in the level file, controlled in `final_level_counter.c`. In Ice Palace, the bird monster is hard-coded to end the level when it reaches the final frame of its death anim. Putting the bird monster in other levels has the same effect, regardless of the script (should become optional IMO for custom levels, but needs data injection to make OG work, so a future task - I'll raise a ticket as a reminder for when we get to that stage).

I imagine we can properly verify they're not in use after full decomp, then later look at tidying up.

File: src/game/gameflow.c [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+            }
+            ptr += 2;
+            break;
+
+        case GFE_ADD_TO_INV:
+            if (type != GFL_STORY && type != GFL_MIDSTORY) {
+                if (ptr[1] < 1000) {
+                    g_GF_SecretInvItems[ptr[1]]++;
+                } else if (type != GFL_SAVED) {
+                    g_GF_Add2InvItems[ptr[1]]++;
```
Comment (lahm86) [inline src/game/gameflow.c]: I think this needs to be `g_GF_Add2InvItems[ptr[1] - 1000]++;`. This is used at the start of Tibetan Foothills and Barkhang Monastery to add The Seraph (puzzle4) to the inventory as it's carried over from The Deck. It's also used in HSH to give `key1`. Currently neither happens.

**Review by aredfan — APPROVED** [review]
I just finished going through the list except FMVs. LGTM.

I should mention that FMVs doesn't work well for me because it causes my pc to become unresponsive. I can hear the FMVs play but the screen goes black, probably caused by windows resizing. It's been the case since 0.1 so nothing to do with this PR.

The game works fine when I exclude the FMV folder from the TR2X directory.

File: src/game/gameflow.c (hunk 1 above)
Comment (rr-) [reply src/game/gameflow.c]: Excellent catch! Fixed: https://github.com/LostArtefacts/TR2X/compare/34dc3b1807bfd929ac08a11feb1da1a0f5476c41..5a182c9b0519b48bd6f66d9de6c88d4f05ed7efd

New test build: https://home.wind.garden/serve/pub/TR2X-0.2-309-g5a182c9-dirty-Windows.zip

**Review by lahm86 — APPROVED** [review]

**Pull request merged**